collector/src/main/java/com/bitmovin/analytics/utils/HttpClient.java

```diff
@@ -19,15 +22,26 @@
     private static final String TAG = "HttpClient";
     private OkHttpClient client = null;
     private Context context;
+    private BitmovinAnalyticsConfig analyticsConfig;
 
-    public HttpClient(Context context) {
+    public HttpClient(Context context, BitmovinAnalyticsConfig analyticsConfig) {
         this.context = context;
+        this.analyticsConfig = analyticsConfig;
     }
 
 
     public void post(String url, String postBody, final Callback callback) {
         if (client == null) {
-            client = new OkHttpClient();
+            if (analyticsConfig.isResendDataOnHttpTimeout()){
+                client = new OkHttpClient.Builder()
+
+                        .retryOnConnectionFailure(false)
+                        .connectTimeout(15, TimeUnit.SECONDS)
```

Comment:
I think defaults are 10s. This client has its own retry process so it will try to resend failed requests. We need at least `connectTimeout(15, TimeUnit.SECONDS)` setting in combination with `retryOnConnectionFailure` to disable its own retry process. Otherwise I see duplicated samples in DB :(